gravitee-apim-console-webui/src/management/api/portal/userGroupAccess/groups/api-portal-access-groups.component.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { switchMap, takeUntil, tap } from 'rxjs/operators';
+import { EMPTY, Subject } from 'rxjs';
+
+import { UIRouterStateParams } from '../../../../../ajs-upgraded-providers';
+import { GroupService } from '../../../../../services-ngx/group.service';
+import { Group } from '../../../../../entities/group/group';
+import { ApiService } from '../../../../../services-ngx/api.service';
+import { GioPermissionService } from '../../../../../shared/components/gio-permission/gio-permission.service';
+import { SnackBarService } from '../../../../../services-ngx/snack-bar.service';
+
+@Component({
+  selector: 'api-portal-access-groups',
+  template: require('./api-portal-access-groups.component.html'),
+  styles: [require('./api-portal-access-groups.component.scss')],
+})
+export class ApiPortalAccessGroupsComponent implements OnInit, OnDestroy {
+  private unsubscribe$: Subject<void> = new Subject<void>();
+
+  public isReadOnly = true;
+  public form: FormGroup;
+  public selectedGroups: FormControl = new FormControl({ value: [], disabled: this.isReadOnly });
+  public groups: Group[];
+  public initialUserGroupIds: string[];
+
+  constructor(
+    @Inject(UIRouterStateParams) private readonly ajsStateParams,
+    private readonly apiService: ApiService,
+    private readonly groupService: GroupService,
+    private readonly permissionService: GioPermissionService,
+    private readonly snackBarService: SnackBarService,
+    private readonly formBuilder: FormBuilder,
+  ) {
+    this.form = this.formBuilder.group({});
+  }
+
+  ngOnInit() {
+    this.isReadOnly = !this.permissionService.hasAnyMatching(['api-definition-u']);
+    this.groupService
+      .list()
+      .pipe(
+        switchMap((groups) => {
+          this.groups = groups;
+          return this.apiService.get(this.ajsStateParams.apiId);
+        }),
+        takeUntil(this.unsubscribe$),
```

Comment:
it seems to me :  it should be the first of the pipe elements.
because the order is important. if apiService.get takes too long and the order of unsubscribe arrives it is not stop immediately.  
unless I'm wrong, I'm not so sure anymore.